handlers.go

```diff
@@ -158,6 +158,101 @@ func UpsertReputationByViolationHandler(w http.ResponseWriter, r *http.Request)
 	}
 }
 
+func writeEntryErrorResponse(w http.ResponseWriter, errno int, entryIndex int, entry IpViolationEntry, msg string) {
+	type EntryError struct {
+		Errno int
+		EntryIndex int
+		Entry IpViolationEntry
+		Msg string
+	}
+
+	entryError := EntryError{
+		Errno: errno,
+		EntryIndex: entryIndex,
+		Entry: entry,
+		Msg: msg,
+	}
+	j, err := json.Marshal(entryError)
+	if err != nil {
+		log.Warnf("error marshaling error response JSON: %s", err)
+	}
+	w.Header().Set("Content-Type", "application/json")
+
+	w.WriteHeader(http.StatusBadRequest)
+	w.Write(j)
+	return
+}
+
+
+func MultiUpsertReputationByViolationHandler(w http.ResponseWriter, r *http.Request) {
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		log.WithFields(log.Fields{"errno": BodyReadError}).Warnf(DescribeErrno(BodyReadError))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	var entries []IpViolationEntry
+	err = json.Unmarshal(body, &entries)
+	if err != nil {
+		log.WithFields(log.Fields{"errno": JSONUnmarshalError}).Warnf(DescribeErrno(JSONUnmarshalError), err)
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	if len(entries) < 1 {
+		log.WithFields(log.Fields{"errno": MissingIPViolationEntryError}).Warn(DescribeErrno(MissingIPViolationEntryError))
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	if len(entries) > maxEntries {
+		log.WithFields(log.Fields{"errno": TooManyIpViolationEntriesError}).Warn(DescribeErrno(TooManyIpViolationEntriesError))
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	if db == nil {
+		log.WithFields(log.Fields{"errno": MissingDB}).Warnf(DescribeErrno(MissingDB))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	for i, entry := range entries {
```

Comment:
iiuc, this endpoint does the same thing as the `UpsertReputationByViolationHandler`, but for lists of violations instead of single ones. If so, shouldn't they share code?